### ISSUE TITLE
Integrate with RSS-Bridge

### ISSUE THREAD

**Issue by tiotrom** [friendica/friendica#9446]
Friendica is amazing and one of the unique features is that it works with all of the other federated protocols + twitter + rss. It means, I can follow anyone out there (almost), and engage with them. Since it supports RSS, imagine integrating it with the RSS-Bridge https://github.com/RSS-Bridge/rss-bridge (maybe via an addon) so that users can follow facebook pages and profiles, youtube channels, instagram, and so many other sources.

What I love about Friendica is that I can grab a Twitter URL, or mastodon, or RSS, and so on, drop them into the search bar, then I can simply add them as contacts and follow them. Imagine people can do that with youtube, facebook, instagram, and the like.

Does it make sense?

**Comment by tobiasd** [comment]
You mean an addon that adds a RSS-Bridge to your ~friendica node so that one can than follow the RSS feeds generated by it?

**Comment by tiotrom** [comment]
Yup. So that it integrates better with say youtube, instagram, facebook and the like. It detects a youtube url and puts it through the RSS-Bridge to generate an RSS feed that Friendica users can follow.

**Comment by tiotrom** [comment]
Oh I see it grabs the RSS of youtube channels, which is awesome. But not FB pages. Say I add this https://www.facebook.com/tromproject/ to my contacts. It doesn't work, but with the help of RSS-Bridge it could work. That's what I mean.

**Comment by tiotrom** [comment]
I wanted to subscribe to a subreddit for instance, but friendica cannot pull an RSS for that. RSS-Bridge can. And I used that as such https://tromland.org/rss-bridge/?action=display&bridge=Reddit&context=single&r=linux&format=Html. Just saying, would be cool to have this to Friendica, to be like the ultimate connector to everything on the web :).

**Comment by MrPetovan** [comment]
I looked at the project and it definitely looks like something we could provide through an addon.

**Comment by tiotrom** [comment]
That could open up a lot more in terms of compatibility.

**Comment by MrPetovan** [comment]
My main question is whether a bridge URL can be generated automatically from the original website URL. Also from the code itself it looks like we'll have to do quite a bit of code surgery to transplant the bridges themselves since we don't need the whole frontend, and their project isn't available on Packagist (the Composer repository).

**Comment by MrPetovan** [comment]
Hum, the bridge search from an original URL is only done in Javascript to show the associated bridge, this means we'll have to write regular expressions for each bridge to transform any given supported URL into the parameters bridges expect.

**Comment by MrPetovan** [comment]
Never mind, there is a parameter detection feature but it's often missing from bridges. I would still have to implement the detection for the bridges we want to include in the addon.

**Comment by tiotrom** [comment]
Does it help to have the RSS-Bridge implemented on a particular website? Like we have one at https://tromland.org/rss-bridge/? I am thinking, maybe you can parse the links through a running instance of RSS Bridge. We for sure are keeping our instance up and running since we use it for our own news system.

**Comment by MrPetovan** [comment]
So, here's how I imagine this prospective addon to work for users:
- The addon will implement a hook on the contact add module and run the expected contact URL through the available bridge detection functions.
- If a match is found, it will generate a URL looking like `https://<node domain>/rssbridge?url=<original URL>` which will use the RSS bridge function to output a syndication feed.
- This feed will be heavily cached (12 or even 24 hours) to prevent nodes from polling the remote websites too often. This will make the polling frequency setting of the created feed contacts moot.
- The generated local URL will be used for the feed contact since we can't differentiate between a contact URL and a contact homepage.
- This addon may also implement a hook on the general search and perform the RSS Bridge detection on anything looking like a URL.

And for admin:
- Allowed bridges will be able to be chosen in a checkbox list at the node level.

**Comment by tiotrom** [comment]
In layman's words, it means that one can copy paste a facebook page url to the search/contact and be able to follow them like with any other RSS feed? Same with Instagram accounts, youtube, and the like?

**Comment by MrPetovan** [comment]
Yes, this will be how it works. Unfortunately, I don't think it will be possible to link to the original page, I have to do actual tests.

**Comment by tiotrom** [comment]
Well I think this will make Friendica even more awesome! I don't personally care about facebook, instagram and the like, but I would love to see people being able to follow pages/people from such well known networks on Friendica since it can make the transition to Friendica easier.
